050-AIS-APIManagementwithFunctions/Student/Challenge-01.md

```diff
@@ -0,0 +1,91 @@
+
+## Pre-requisites
+
+- You should have completed Challenge 00
+
+## Introduction
+You are a cloud engineer for Runva - a global health and wellness company that provides solutions for individuals who takes running as sport, providing them a platform to track running activities, analyze performance and recommend plans to help them train smarter and safer.  They are going to embark on a six-month project to modernize their integration solution hosted on-premises and migrate them to Azure integration services.  Since this is their first foray in Azure, you are tasked to create a proof-of-concept (POC) that captures several integration patterns.
+
+
+## Description
+As a cloud engineer, you would like to be able to deploy this POC environment using Infrastructure-as-Code.  You have learned that you can efficiently do this in Azure using the new language called Bicep. As a baseline, you are going to deploy API Management service as gateway for your APIs, as well as a Function App where you would host your APIs.
+
+There are two scenarios you would like to prove (just choose one):
+
+### Scenario1: Deploy a VNET-secured AIS environment 
+This scenario is for when you need to access services deployed over a private network (e.g. APIs hosted on-premises)
+- Deploy the Bicep templates of the VNET-integrated AIS which can be found at /Challenge-01/Scenario-01 of the Resources.zip file provided to you by your coach.  There are some missing parameter or variable values that you need to fill out, and make sure that you fix all the errors and warnings before deploying.
+
+### Scenario2: Deploy an identity-secured AIS environment 
+This scenario is preferred for integrating with services hosted in the cloud or publicly-accessible
+- Deploy the Bicep templates of the publicly-exposed AIS which can be found at /Challenge-01/Scenario-02 of the Resources.zip file provided to you by your coach.  There are some missing parameter or variable values that you need to fill out, and make sure that you fix all the errors and warnings before deploying.
+    - There should be several [Bicep modules](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/modules) for deploying individual resources, the file structure should be as follows:
+      - main.bicep
+        - modules
+          - functions.bicep
+          - apim.bicep
+          - appinsights.bicep
+    - The resources should have the following properties at a minimum:
+      - name
+      - location
+      - sku/kind
+      - resource-specific properties
+    - Function app and APIM needs to configured with Application insights. Therefore, ensure that you create this resource first, and make sure that you define [output parameters](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/outputs?tabs=azure-powershell) for instrumentation key and resource id.  You would need to pass values as input to the Function App and APIM modules. 
```

Comment:
Grammar - Needs to be configured with Application Insights

Reply:
Done